Airport Ticket Booking System/Repositories/BookingRepository.cs

```diff
@@ -0,0 +1,151 @@
+﻿namespace Airport_Ticket_Booking_System;
+
+public  class BookingRepository : IBookingRepository
+{
+    private readonly List<Booking> _bookings = new List<Booking>();
+    private const string CSVFilePath = "bookings.csv";
+    private readonly object _lock = new object();
+
+    public BookingRepository()
+    {
+        LoadBookingsFromCSVFileAsync().Wait();
+    }
+
+    public async Task AddBookingAsync(Booking booking)
+    {
+        lock (_lock)
+            _bookings.Add(booking);
+
+        await SaveBookingsToCSVFileAsync();
+        Console.WriteLine("Booking added successfully.");
+    }
+
+    public async Task CancelBookingAsync(string bookingID)
+    {
+        lock (_lock)
+        {
+            var booking = _bookings.FirstOrDefault(b => b.bookingID == bookingID);
+            if (booking != null)
+            {
+                _bookings.Remove(booking);
+            }
+            else
+            {
+                Console.WriteLine($"Booking with ID: {bookingID} not found.");
+                return;
+            }
+        }
+
+        await SaveBookingsToCSVFileAsync();
+        Console.WriteLine($"Booking with ID: {bookingID} has been canceled.");
+    }
+
+    private async Task LoadBookingsFromCSVFileAsync()
+    {
+        if (File.Exists(CSVFilePath))
```

Comment:
NIT: you can just do an early return instead, so you don't have many nested blocks:

```csharp
    if (!File.Exists(CSVFilePath))
        return
    /// return of your code
```